inc/class-yoast-acf-analysis.php

```diff
@@ -30,69 +30,65 @@ public function admin_init() {
 
 		$this->boot();
 
-			if ( defined( 'YOAST_ACF_ANALYSIS_ENVIRONMENT' ) && 'development' === YOAST_ACF_ANALYSIS_ENVIRONMENT ) {
-				$this->boot_dev();
-			}
+		if ( defined( 'YOAST_ACF_ANALYSIS_ENVIRONMENT' ) && 'development' === YOAST_ACF_ANALYSIS_ENVIRONMENT ) {
+			$this->boot_dev();
 		}
 
+		$this->register_config_filters();
+
 		$assets = new Yoast_ACF_Analysis_Assets();
 		$assets->init();
 	}
 
 	/**
-	 * Boot the plugin
+	 * Boots the plugin.
 	 */
 	public function boot() {
 
-		if ( is_null( Yoast_ACF_Analysis_Registry::instance()->get( 'config' ) ) ) {
-
-			$default_configuration = new Yoast_ACF_Analysis_Configuration_Default( $this->get_blacklist(), $this->get_field_selectors() );
-
-			$configuration = apply_filters(
-				Yoast_ACF_Analysis_Configuration::PLUGIN_NAME . '/config',
-				$default_configuration
-			);
-
-			if ( ! ($configuration instanceof Yoast_ACF_Analysis_Configuration) ) {
-				$configuration = $default_configuration;
-			}
-
-			Yoast_ACF_Analysis_Registry::instance()->add( 'config', $configuration );
-
+		$registry = Yoast_ACF_Analysis_Facade::get_registry();
+		if ( null !== $registry->get( 'config' ) ) {
```

Comment:
I still think that an interface would be cleaner. I mean if you don't want to reimplement it from scratch you can still just extend the default implementation. But if you really want to you can still roll your own implementation. On the other hand I don't see what harm is done using an interface here.

Also the way to code is now you can still replace the configuration class with a custom implementation because there is [no check for the right class here](https://github.com/Yoast/yoast-acf-analysis/blob/292c0bc339e5c3544ba8c8def92a064ad85f6002/inc/class-yoast-acf-analysis.php#L48).

So this should probably be something like

`if ( null !== $registry->get( 'config' ) && $registry->get( 'config' ) instanceof Yoast_ACF_Analysis_Configuration ) {`